cli/create.go

```diff
@@ -446,6 +431,74 @@ type prepWorkspaceBuildArgs struct {
 	RichParameterDefaults []codersdk.WorkspaceBuildParameter
 }
 
+// resolvePreset determines which preset to use based on the --preset flag,
+// or prompts the user to select one if the flag is not provided.
+func resolvePreset(inv *serpent.Invocation, presets []codersdk.Preset, presetName string) (*codersdk.Preset, error) {
+	// If preset name is specified, find it
+	if presetName != "" {
+		for _, preset := range presets {
+			if preset.Name == presetName {
+				return &preset, nil
+			}
+		}
+		return nil, xerrors.Errorf("preset %q not found", presetName)
+	}
+
+	// No preset specified, prompt user to select one
+	return promptPresetSelection(inv, presets)
```

Comment:
nonblocking: chaining this call to promptPresetSelection here has a few coupling related downsides:
* It locks any use and testing of this parameter resolution into preset prompting as well.
* It requires an additional parameters (inv) that is not relevant to the main body of this function and is only passed through.

If possible, it would perhaps be better to return something here that means "no preset resolved" and then call the prompt outside.